hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3053,21 +3053,22 @@ private void scan(HBaseRpcController controller, ScanRequest request, RegionScan
           builder.setScanMetrics(metricBuilder.build());
         }
       }
+    } finally {
+      region.closeRegionOperation();
+      // Update serverside metrics, even on error.
       long end = EnvironmentEdgeManager.currentTime();
       long responseCellSize = context != null ? context.getResponseCellSize() : 0;
       region.getMetrics().updateScanTime(end - before);
       if (regionServer.metricsRegionServer != null) {
-        regionServer.metricsRegionServer.updateScanSize(
-            region.getTableDesc().getTableName(), responseCellSize);
-        regionServer.metricsRegionServer.updateScanTime(
-            region.getTableDesc().getTableName(), end - before);
+        regionServer.metricsRegionServer.updateScanSize(region.getTableDesc().getTableName(), responseCellSize);
+        regionServer.metricsRegionServer.updateScanTime(region.getTableDesc().getTableName(), end - before);
+        regionServer.metricsRegionServer
+          .updateReadQueryMeter(region.getRegionInfo().getTable(), numOfNextRawCalls);
+      }
+      // coprocessor postNext hook
+      if (region.getCoprocessorHost() != null) {
```

Comment:
I remember branch-2/master didn't have this change?

Reply:
Right, branch-2/master don't have this change yet.

Reply:
So, why do we need this change

Reply:
Still, I got no answer about this 'why'. Could you remove this
`if (region.getCoprocessorHost() != null) {` branch out of block, and stick to the parent task's pull request's changes.